MCPClient: prevent archivematicaUpdateSizeAndChecksum.py from detecting OCR transcription files as derivatives

jrwdunham opened the pull request:
Fixes #1043 

Requires https://github.com/artefactual-labs/mets-reader-writer/pull/46

sevein reviewed (approved):
Nicely done, the interface feels great too!

jrwdunham committed:
MCPClient: OCR transcrs are not derivs on reingest

Stops archivematicaUpdateSizeAndChecksum.py from treating OCR
transcription files as derivatives on full re-ingest. Because OCR files
are deleted on full re-ingest, the previous behaviour was causing a
failure when OCR transcriptions were documented in the METS file but
preservation derivatives were not. The presence of preservation
derivatives (which are not deleted form the Files table) was masking this issue
in typical transfers. See #1043.

Fixes #1043

jrwdunham commented:
Cherry-picked to qa/1.x as 822bfdd58b33d4902354c38c5ae24274e84cfa89
